### ISSUE TITLE
Introduce multishard_writer

### ISSUE THREAD

**Issue by asias** [scylladb/scylla#3497]
We need a multishard_writer which gets mutation fragments from a producer (e.g., from the network using the rpc streaming) and consumes the mutation fragments with a consumer (e.g., write to sstable).

The multishard_writer will take care of the mutation fragments do not belong to current shard. 

This multishard_writer will be used in the new scylla streaming.

**Comment by duarten** [comment]
This is a weird issue. It doesn't seem to report a problem or shortcoming, but to expound the implementation details on some non-existing class. Can you close it please?

**Comment by asias** [comment]
It is used to track a new class which will be used in the new streaming code.  We need a issue number to backport.  It will be closed when the patch is merged.

**Comment by duarten** [comment]
This is not an issue, it's a set of implementation details. Why would we want to backport this class? Please either close it, or reword it in understandable terms.

**Comment by asias** [comment]
We use the issue system to track existing issues and new features. The new multishard_writer class is part of the new streaming.

If you do not like the details. I can get rid of it.

**Comment by duarten** [comment]
Ah, then the issue should be about the new streaming, not about an implementation detail of the new streaming. That would be the issue mentioned in a potential backport, and the multishard_writer patch would just be a dependency.

**Comment by asias** [comment]
new streaming will have its own issue.

**Comment by asias** [comment]
And this new class can also be used in new repair. 

Why would I get rid of this issue?

**Comment by duarten** [comment]
Because it is not an issue, it doesn't describe a problem or shortcoming. It describes a helper class of the new streaming or repair code, which actually solve a particular problem.

**Comment by asias** [comment]
The problem is we do not have such class which is needed for improvement for streaming and repair.  
We need someone to work on this issue. Shlomi can assign this issue to someone.

**Comment by asias** [comment]
I do not understand the point of arguing this. @avikivity @slivne since when we are not allowed to create new issue for development/track a new piece of code?

**Comment by duarten** [comment]
The updated description makes a bit more sense, but I think you missed my point. If there's a patch on the mailing list adding a piece of infrastructure to help with real issues, there's no need to open an issue in the _bug tracker_ with details about the contents of the patch (after the patch has been sent), because when backporting, said patch will be picked up as a dependency of the real, underlying issues.

**Comment by asias** [comment]
> The updated description makes a bit more sense, but I think you missed my point. If there's a patch on the mailing list adding a piece of infrastructure to help with real issues, there's no need to open an issue in the bug tracker with details about the contents of the patch (after the patch has been sent), because when backporting, said patch will be picked up as a dependency of the real, underlying issues.

Creating a new issue is not just for backporting. Tracking the development is another reason.

**Comment by asias** [comment]
On Tue, Jun 19, 2018, 13:23 Duarte Nunes <notifications@github.com> wrote:

> I don’t recall writting that opening issues was just for baclporting; that
> doesn’t make any sense.
>

You wrote I should close this issue because when we backport we can use
another issue number e.g., the one for streaming.

I wrote this issue is created not just for backporting.

I wrote that this is a bug tracker used to track bugs or enhancements, not
> to track internal implementation details.
>
There is no  details about the implementation any more. I never intended to
do so even if there was some details about the implement in the original
issue report.

It it were, all patches would close an issue, which is clearly not the case.
